plugins/extractors/metabase/extractor.go

```diff
@@ -0,0 +1,160 @@
+package metabase
+
+import (
+	"bytes"
+	"context"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strconv"
+	"time"
+
+	_ "github.com/lib/pq"
+	"github.com/odpf/meteor/core"
+	"github.com/odpf/meteor/core/extractor"
+	"github.com/odpf/meteor/plugins"
+	"github.com/odpf/meteor/proto/odpf/meta"
+	"github.com/odpf/meteor/utils"
+)
+
+var (
+	client = &http.Client{
+		Timeout: 2 * time.Second,
+	}
+)
+
+type Config struct {
+	UserID    string `mapstructure:"user_id" validate:"required"`
+	Password  string `mapstructure:"password" validate:"required"`
+	Host      string `mapstructure:"host" validate:"required"`
+	SessionID string `mapstructure:"session_id"`
+}
+
+type Extractor struct {
+	cfg       Config
+	sessionID string
+	logger    plugins.Logger
+}
+
+func New(logger plugins.Logger) *Extractor {
+	return &Extractor{
+		logger: logger,
+	}
+}
+
+// Extract collects metdata from the source. Metadata is collected through the out channel
+func (e *Extractor) Extract(ctx context.Context, config map[string]interface{}, out chan<- interface{}) (err error) {
+	// Build and validate config received from receipe
+	if err := utils.BuildConfig(config, &e.cfg); err != nil {
+		return extractor.InvalidConfigError{}
+	}
+	// get session id for further api calls in metabase
+	if e.cfg.SessionID == "" {
```

Comment:
since we already have `getSessionID` method, we could move this `if` logic to that function instead.
it should clean a few lines in this function.

Reply:
So we should call function ```getSessionID``` directly and make the session id's request as per this condition??

Reply:
yeah basically `getSessionID` will be responsible for creating sessionID. This function will just call it and expect a sessionID without caring how to generate it.

Reply:
i would something like
```
e.sessionID = e.getSessionID()
```